src/test/java/com/the/dark/side/crew/fejsbuk/mapper/CommentMapperTest.java

```diff
@@ -0,0 +1,133 @@
+
+class CommentMapperTest {
+
+    @InjectMocks
+    private CommentMapper commentMapper;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private PostRepository postRepository;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void whenCallToDtosThenReturnListOfDtos() {
+        CommentMapper mapper = spy(new CommentMapper(userRepository, postRepository));
+        CommentDto commentDto = mock(CommentDto.class);
+        CommentEntity commentEntity = mock(CommentEntity.class);
+
+        doReturn(commentDto).when(mapper).toDto(commentEntity);
+
+        List<CommentDto> result = mapper.toDtos(List.of(commentEntity));
+        assertEquals(1, result.size());
+        assertEquals(commentDto, result.get(0));
+    }
+
+    @Test
+    void whenCallToDtoThenReturnPostDto() {
+        CommentEntity commentEntity = mock(CommentEntity.class);
+        PostEntity postEntity = mock(PostEntity.class);
+        UserEntity userEntity = mock(UserEntity.class);
+        LocalDateTime createdAt = LocalDateTime.now();
+        long commentId = 3L;
+        long postId = 1L;
+        long userId = 2L;
+        String message = "message";
+
+        when(commentEntity.getId()).thenReturn(commentId);
+        when(commentEntity.getMessage()).thenReturn(message);
+        when(commentEntity.getCreatedAt()).thenReturn(createdAt);
+        when(commentEntity.getUserEntity()).thenReturn(userEntity);
+        when(commentEntity.getPostEntity()).thenReturn(postEntity);
+        when(userEntity.getId()).thenReturn(userId);
+        when(postEntity.getId()).thenReturn(postId);
+
+        CommentDto result = commentMapper.toDto(commentEntity);
+        assertEquals(commentId, result.getId());
+        assertEquals(message, result.getMessage());
+        assertEquals(createdAt, result.getCreatedAt());
+        assertEquals(postId, result.getPostId());
+        assertEquals(userId, result.getUserId());
+    }
+
+    @Test
+    void whenUserAndPostFoundThenReturnComment() {
+        long userId = 1L;
+        long postId = 2L;
+        String message = "message";
+        UserEntity userEntity = mock(UserEntity.class);
+        PostEntity postEntity = mock(PostEntity.class);
+        CommentDto commentDto = mock(CommentDto.class);
+
+        when(commentDto.getMessage()).thenReturn(message);
+        when(commentDto.getUserId()).thenReturn(userId);
+        when(commentDto.getPostId()).thenReturn(postId);
+        when(userRepository.findById(userId)).thenReturn(Optional.of(userEntity));
+        when(postRepository.findById(postId)).thenReturn(Optional.of(postEntity));
+        when(userEntity.getId()).thenReturn(userId);
+        when(postEntity.getId()).thenReturn(postId);
+
+        CommentEntity result = commentMapper.toEntity(commentDto);
+        assertEquals(message, result.getMessage());
+        assertEquals(userId, result.getUserEntity().getId());
```

Comment:
Just check if it's user which u expected

Reply:
I don't understand, in PostMapperTest is the same situation and it's ok.